controllers/hrController.ts

```diff
@@ -0,0 +1,259 @@
+import { NextFunction, Request, Response } from 'express';
+import { UserDb } from '../models/UserSchema';
+import { UserSkillDb } from '../models/UserSkillsSchema';
+import { Role, Status, UserSkillsExpectations } from '../types';
+import { filterHr, userInfo, userProfile, userToHrResponse } from '../utils/filterResponse';
+import { pagination, pipeline, getAvailableUsers, getReservedUsers } from '../utils/pagination';
+import { handleEmail } from '../utils/handleEmail';
+import { employedEmailTemplate } from '../templates/employedEmailTemplate';
+import { ValidationError } from '../utils/handleError';
+
+export const availableUsers = async (req: Request, res: Response, next: NextFunction) => {
+  const page = parseInt(req.query.page as string) || 1;
+  const limit = parseInt(req.query.limit as string) || 10;
+
+  try {
+    const { results, totalCount, totalPages } = await pagination(
+      pipeline({ email: { $in: await getAvailableUsers() } }),
+      page,
+      limit
+    );
+
+    const availableUsersProfile = results
+      .map((user): UserSkillsExpectations => {
+        if (user.profile && user.profile._id !== null) {
+          return userToHrResponse(user);
+        }
+        return null;
+      })
+      .filter((user) => user);
+
+    res.status(200).send({ users: availableUsersProfile, totalCount, totalPages });
+  } catch (err) {
+    next(err);
+  }
+};
+
+export const searchUsers = async (req: Request, res: Response, next: NextFunction) => {
+  const { id } = req.params;
+  const page = parseInt(req.query.page as string) || 1;
+  const limit = parseInt(req.query.limit as string) || 10;
+  const search = req.query.search || '';
+  const tab = req.query.tab as string;
+
+  if (!['1', '2'].includes(tab)) throw new ValidationError('Nieprawidłowy parametr zakładki.');
+
+  const filter = {
+    $or: [
+      { email: { $regex: `^.*${search}.*$`, $options: 'i' } },
+      { 'profile.firstName': { $regex: `^.*${search}.*$`, $options: 'i' } },
+      { 'profile.lastName': { $regex: `^.*${search}.*$`, $options: 'i' } },
```

Comment:
Do weryfikacji na ile to bezpieczne. Macie jakąś weryfikacja danych przychodzących? Żeby ktoś tu nie zrobił ReDoSa.